openfeature/multiprovider/strategies.go

```diff
@@ -0,0 +1,154 @@
+package multiprovider
+
+import (
+	"context"
+	"maps"
+	"regexp"
+	"strings"
+
+	of "github.com/open-feature/go-sdk/openfeature"
+)
+
+const (
+	// Strategies
+
+	// StrategyFirstMatch First provider whose response that is not FlagNotFound will be returned. This is executed
+	// sequentially, and not in parallel.
+	StrategyFirstMatch EvaluationStrategy = "strategy-first-match"
+	// StrategyFirstSuccess First provider response that is not an error will be returned. This is executed sequentially
+	StrategyFirstSuccess EvaluationStrategy = "strategy-first-success"
+	// StrategyComparison All providers are called in parallel. If all responses agree the value will be returned.
+	// Otherwise, the value from the designated fallback provider's response will be returned. The fallback provider
+	// will be assigned to the first provider registered.
+	StrategyComparison EvaluationStrategy = "strategy-comparison"
+	// StrategyCustom allows for using a custom Strategy implementation. If this is set you MUST use the WithCustomStrategy
+	// option to set it
+	StrategyCustom EvaluationStrategy = "strategy-custom"
+
+	ReasonAggregated         of.Reason = "AGGREGATED"
+	ReasonAggregatedFallback of.Reason = "AGGREGATED_FALLBACK"
+
+	ErrAggregationNotAllowedText = "object evaluation not allowed for non-comparable types without custom comparable func"
```

Comment:
This should be in its own `const ()` block since it's not an [EvaluationStrategy].